### PR TITLE
New wizard spell #1: Heal Other

### PR DESCRIPTION
**Pull request by ghost**
Adds a new spell for wizards: Heal Other. As the name implies, it cannot heal the caster.
The spell has a 30 second cooldown and has two reductions (7.5 seconds each) to 15 seconds minimum. There are also two separate upgrades: one which allows the spell to affect a 3x3 AoE, and one which gives the spell a 50% chance per organ of healing bleeding/broken bones/organ damage. The latter upgrade will only affect the primary target even if you purchase both.

:cl:
 * rscadd: New healing spell added for wizards.

### CODE REVIEW

**Comment by Dacendeth** [comment]
Seems like something that would be fitting to also be made into a specialization for an apprentice.

**Comment by ghost** [comment]
That's definitely an idea that's on the table, but I didn't want to include it in this PR since I'm not quite sure how to balance it yet.

**Comment by SonixApache** [comment]
reminder that we have a rod of healing sprite in the code, and that there used to be an apprentice subtype specifically for healing

**Comment by Caboozles** [comment]
Does this affect mechanical organs? Because the fucking shenanigans with EMP and this could be hilarious

**Comment by Caboozles** [comment]
Does this affect mechanical organs? Because the fucking shenanigans with EMP and this could be hilarious

**Comment by ghost** [comment]
It should work on mechanical organs since they behave the same as regular ones with any special interactions added on top.

**Comment by SonixApache** [comment]
be sure to add a spellbook

**Comment by ghost** [comment]
> 
> 
> be sure to add a spellbook

Not adding a spellbook for this one because it's niche and would drag down bundles as a result. Maybe for the other one, though.

File: code/modules/spells/targeted/heal.dm [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	specialization = UTILITY
+	spell_levels = list(Sp_SPEED = 0, Sp_POWER = 0, Sp_RANGE = 0)
+
+	school = "transmutation"
+	charge_max = 300
+	cooldown_reduc = 75
+	cooldown_min = 150
+	invocation = "DI TIUB SEEL IM"
+	invocation_type = SpI_SHOUT
+	message = "<span class='danger'>You feel refreshed.<span>"
```
Comment (Kurfursten) [inline code/modules/spells/targeted/heal.dm]: span class 'good', 'info', or 'sinister' seems a better fit

File: code/modules/spells/targeted/heal.dm [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	amt_dam_tox = -15
+
+	spell_flags = WAIT_FOR_CLICK
+
+	hud_state = "wiz_heal"
+
+/spell/targeted/heal/cast(var/list/targets, mob/user)
+	for(var/mob/living/T in targets)
+		if(spell_levels[Sp_RANGE])
+			for(var/mob/living/M in range(1, T))
```
Comment (Kurfursten) [inline code/modules/spells/targeted/heal.dm]: can you make sure this doesn't grab the original target a second time?

also, interestingly, this upgraded spell might want the ability to target a turf (imagine if you will 2 players standing with 1 tile between them, you could click that tile to heal both with AOE)

File: code/modules/spells/targeted/heal.dm (hunk 1 above)
Comment (ghost) [reply code/modules/spells/targeted/heal.dm]: Good catch. I changed it to sinister.

File: code/modules/spells/targeted/heal.dm (hunk 2 above)
Comment (ghost) [reply code/modules/spells/targeted/heal.dm]: Made sure it doesn't stack with the original target. Not sure about making it target a turf, you can already AoE heal two people by just clicking one of them.

**Pull request merged**